File: add-buy-links.py
import json
import urllib.parse

# Read the components database
with open('data/components.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

def generate_amazon_link(name, category=''):
    """Generate Amazon India search link for a product"""
    # Clean up the name for better search results
    search_term = name.replace(' / ', ' ').replace('/', ' ')
    # Add category context for better results
    if category:
        search_term = f"{search_term} {category}"
    encoded = urllib.parse.quote_plus(search_term)
    return f"https://www.amazon.in/s?k={encoded}"

# Add buy links to all categories
categories = {
    'laptops': 'laptop',
    'cpus': 'processor',
    'gpus': 'graphics card',
    'mobos': 'motherboard',
    'ram': 'RAM memory',
    'storage': 'SSD',
    'psu': 'power supply',
    'case': 'PC case'
}

total_added = 0
for category_key, category_name in categories.items():
    if category_key in data:
        for item in data[category_key]:
            if 'buyLink' not in item or not item.get('buyLink'):
                item['buyLink'] = generate_amazon_link(item['name'], category_name)
                total_added += 1
                print(f"Added link for {category_key}: {item['name'][:50]}...")

# Save the updated database
with open('data/components.json', 'w', encoding='utf-8') as f:
    json.dump(data, f, indent=2, ensure_ascii=False)

print(f"\n✅ Successfully added {total_added} Amazon buy links!")
print(f"📦 Total products in database: {sum(len(data[cat]) for cat in categories.keys() if cat in data)}")

File: backend/add-buy-links.js
const fs = require('fs');
const path = require('path');

// Read the components database
const data = JSON.parse(fs.readFileSync(path.join(__dirname, '..', 'shared', 'data', 'components.json'), 'utf8'));

function generateAmazonLink(name, category = '') {
    // Clean up the name for better search results
    let searchTerm = name.replace(/ \/ /g, ' ').replace(/\//g, ' ');
    // Add category context for better results
    if (category) {
        searchTerm = `${searchTerm} ${category}`;
    }
    const encoded = encodeURIComponent(searchTerm);
    return `https://www.amazon.in/s?k=${encoded}`;
}

// Add buy links to all categories
const categories = {
    'laptops': 'laptop',
    'cpus': 'processor',
    'gpus': 'graphics card',
    'mobos': 'motherboard',
    'ram': 'RAM memory',
    'storage': 'SSD',
    'psu': 'power supply',
    'case': 'PC case'
};

let totalAdded = 0;
let totalProducts = 0;

for (const [categoryKey, categoryName] of Object.entries(categories)) {
    if (data[categoryKey]) {
        for (const item of data[categoryKey]) {
            totalProducts++;
            if (!item.buyLink || item.buyLink === '') {
                item.buyLink = generateAmazonLink(item.name, categoryName);
                totalAdded++;
                console.log(`Added link for ${categoryKey}: ${item.name.substring(0, 50)}...`);
            }
        }
    }
}

// Save the updated database
fs.writeFileSync('data/components.json', JSON.stringify(data, null, 2), 'utf8');

console.log(`\n✅ Successfully added ${totalAdded} Amazon buy links!`);
console.log(`📦 Total products in database: ${totalProducts}`);
console.log(`🔗 Products with buy links: ${totalProducts}`);
